src/main/java/com/app/models/MemberModel.java

```diff
@@ -1,3 +1,146 @@
 package com.app.models;
 
-public class MemberModel {}
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+public class MemberModel {
+  private String ID;
+  private String name;
+  private LocalDate birthdate;
+  private GenderModel gender;
+  private String phoneNumber;
+  private boolean competitive;
+  private ArrayList<DisciplineModel> disciplines = new ArrayList<>();
+  private final LocalDate creationDate;
+  private ArrayList<MembershipModel> memberships = new ArrayList<>();
+
+  public MemberModel() {
+    creationDate = LocalDateTime.now().toLocalDate();
+  }
+
+  /**
+   * Package Private constructor for instantiating objects in the model.
+   *
+   * <p>Can be used when reading stored data.
+   *
+   * @param ID
+   * @param name
+   * @param birthdate
+   * @param gender
+   * @param phoneNumber
+   * @param competitive
+   * @param disciplines
+   * @param creationDate
+   * @param memberships
+   */
+  MemberModel(
+      String ID,
+      String name,
+      LocalDate birthdate,
+      GenderModel gender,
+      String phoneNumber,
+      boolean competitive,
+      ArrayList<DisciplineModel> disciplines,
+      LocalDate creationDate,
+      ArrayList<MembershipModel> memberships) {
+    setID(ID);
+    setName(name);
+    this.birthdate = birthdate;
+    setGender(gender);
+    setPhoneNumber(phoneNumber);
+    setCompetitive(competitive);
+    this.creationDate = creationDate;
+    setDisciplines(disciplines);
+    setMemberships(memberships);
+  }
+
+  public void setID(String ID) {
+    this.ID = ID;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public void setBirthdate(String birthdate) {
```

Comment:
Add javadoc

Reply:
it will be a one liner when formatting is moved somewhere we can access it globally

I am not sure if it needed - i can make it a regular setter instead without logic

Reply:
should be alright now